### ISSUE TITLE
Popups with url "about:blank", can't be blocked

### ISSUE THREAD

**Issue by uBlock-user** [gorhill/uBlock#2094]
**Issue**

Popups with url "about:blank", can't be blocked

**URL** 

http://gorillavid.in/5uc7qnr3382y

**Screenshot**

https://i.gyazo.com/74b19fe3964c2344a34ca160398d1c7b.png

**Steps To Reproduce**

1]Clear all cookies for gorillavid.in

2]Visit http://gorillavid.in/5uc7qnr3382y

3]Click continue and a popup will appear originating from about:blank and then will be redirected to a data:url which will load because data urls can't be blocked in the same tab[1] and will redirect you further to the popup website before getting killed by uBO. 

**Settings**
All default

- Browser/version: Chromium 56
- uBlock Origin version: 1.9.10

**Filter-Lists**

Default + |about:blank^$popup

**Custom-Filters**

None

**Expected Scenario** uBlock will kill the new tab popup from opening before it proceeds to load

**Actual Scenario** uBlock fails to kill the new tab popup from opening which appears from the URL - about:blank and then gets redirected to a data: url which further tries to load the popup and then uBlock kills the new tab.

[1]https://github.com/gorhill/uBlock/issues/2031

**NOTE - To exploit [1], "about:blank" is used initially, instead of firing a popup from data url directly.**

**Comment by gorhill** [comment]
You are confused, this has nothing to do with #2031. Popups with `data:` can be blocked.

This is not a good idea to block `about:blank` popups, there is nothing illegitimate with `about:blank`, this will likely cause a lot of false positives. If you want to blanket-block popups, just use the no-popup-switch.

**Comment by uBlock-user** [comment]
How am I confused ? A popup occurs from about:blank redirects to data url which deciphers the popup url. Yes it does have everything with 2031. Gorillavid used to launch popups from data urls before, but you patched data url popups and now it's using about:blank instead, simple. 

You don't want to patch this up which is your decision, but it has nothing to do with me being confused. If it causes false positives, then that's another issue, but don't lump it with me being confused. I know what I posted here and it's true. 

That no-popup switch does more harm than good and blocks all new tabs almost everytime. I can't use that. This is your decision not to patch it, but you will patch if it starts being abused by every other site. This is just the beginning. Either way, your choice. nobody's talking about any blanket blocking popups, but only from about:blank which are solely being exploited for loading shitty data url popups.

**Comment by gorhill** [comment]
#2031 is about secondary resources using `data:` URI, not about navigating to a `data` URI.

> will be redirected to a data:url which will load because data urls can't be blocked in the same tab

See picture below, the `data:` URI is blocked:

![](https://cloud.githubusercontent.com/assets/16030872/19404503/a7877414-926e-11e6-9307-02dea4d77255.png)

But I do realize there is a `data:` URI appearing in the address bar, yet it is not reported in the logger. I will investigate.

**Comment by gorhill** [comment]
Sorry, there was a edge case where a URL change in the popup tab/window was not detected by uBO -- though the subsequent ones were caught. The `data:text/html$popup` filter will now be correctly applied for that case.

**Comment by gorhill** [comment]
Fixed with 9b57b827d39dcd7836fc77a021d4aa27dbbb0f37.

**Comment by uBlock-user** [comment]
Don't be too quick to dismiss and close cases. I only report if it's a valid one and if I have experienced it myself.

A piece of advice -  Your logger is great and efficient, but it's not perfect yet. Don't rely on it for everything. There are certain events which your logger won't catch. You will have to see it for yourself. Blindly relying on it will cause you to make bad judgement like this one and many others you make sometimes. 

If someone like me, invests his time in reporting a potential bug, please give the case some time. That's all I ask. Don't be too quick to kick it in the can. Thanks for fixing it.

**Comment by lewisje** [comment]
There are way too many dumb users in these Issues to default to believing that they know what they're doing and what they're looking at; your case is known as an *exception*.

**Comment by gorhill** [comment]
I rely a lot on the content of an issue to answer, so your issue was _"popups with url "about:blank", can't be blocked"_, and I declined to have uBO be able to block `about:blank`. That is still the case.

**Comment by uBlock-user** [comment]
Yeah well that's the title I came up with, what else could I have written ? Speaking of content, I did post all the steps to reproduce which would clear any misconception or misunderstanding you would have from that title, if you feel the title is wrong, change it. I came up with that title because that's what I understood and experienced **visually**. 

I'm not the only exception, there are many who know what they're posting. It's true that majority of the ones are found to dumb, but don't lump the rest like that just because few were like that which is unfortunate in your case (sighs).

**Comment by gorhill** [comment]
> what else could I have written ?

**Exactly** what was the issue: _"uBO fails to see and block `data:` URI popups"_, as it's supposed to be able to.
- You wrongly claimed that wasn't possible, referring to #2031. This has been possible since [1.9.8](https://github.com/gorhill/uBlock/releases/tag/1.9.8), #2031 is unrelated.
- Still, popups were blocked on my side with your repro steps -- so even if I agreed with filtering out `about:blank`, I could not reproduce unwanted popups.
- In the worst case there is no filters for the popups a site is launching, it becomes a filter list issue.

Stick to what exactly is the issue, do not open an issue for a _speculated_ solution (blocking `about:blank` popups which I think is not right) to a _speculated_ issue (which you got wrong, #2031 is unrelated).

I am on the receiving end of all reports (code, filter requests, doc requests, support requests), so I will handle them as efficiently as possible time-wise -- I do not need to be lectured on this. This means I will take issues at face value, if you ask _"uBO should block `about:blank` popups"_, I will still refuse on technical ground unless there are good arguments for this (i.e. _"there is filter in EasyList for this!"_) , and close the issue.

The problem with your issue is that you wrongly speculated uBO was not able to block `data:`-URI popups, and created an issue of what you thought was a solution. I addressed and would still address your solution as 'declined'.

Again for sake of time-efficiency, locking this issue, it has been argued enough.
